hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/RpcConnection.java

```diff
@@ -89,7 +90,18 @@ protected RpcConnection(Configuration conf, HashedWheelTimer timeoutTimer, Conne
       MetricsConnection metrics) throws IOException {
     this.timeoutTimer = timeoutTimer;
     this.codec = codec;
-    this.compressor = compressor;
+    if (compressor != null) {
```

Comment:
nit: no need to check if `compressor` is `null` because you never read it, and subsequent code is apparently fine with a `null` value.

Reply:
Checking if compressor is null is that it can avoid a dns resolution if compressor is null (no compression is configured), which is the common use case.

Reply:
Ah, okay. A comment to this effect would help the casual reader.